spec/vra_spec.rb

```diff
@@ -102,14 +103,43 @@
       expect(state[:resource_id]).to eq('e8706351-cf4c-4c12-acb7-c90cc683b22c')
     end
 
-    describe 'getting the IP address from the server' do
-      context 'when no IP addresses are returned' do
-        it 'raises an exception' do
+    describe 'setting the hostname in the state hash' do
+      context 'when use_dns is true' do
+        let(:config) { { use_dns: true } }
+        it 'raises an exception if the server name is nil' do
+          allow(resource).to receive(:name).and_return(nil)
+          expect { driver.create(state) }.to raise_error(RuntimeError)
+        end
+        it 'uses the server name as the hostname' do
+          driver.create(state)
+          expect(state[:hostname]).to eq('server1')
+        end
+      end
+      context 'when use_dns is false' do
+        it 'raises an exception if no IP address is available' do
           allow(resource).to receive(:ip_addresses).and_return([])
           expect { driver.create(state) }.to raise_error(RuntimeError)
         end
+        it 'uses the IP address as the hostname' do
+          driver.create(state)
+          expect(state[:hostname]).to eq('1.2.3.4')
+        end
       end
+    end
 
+    describe 'getting the IP address from the server when use_dns is true' do
```

Comment:
@stevehedrick we can remove lines 130-149 from spec/vra_spec.rb now, right?  I think they've been replaced by the new tests you wrote in lines 106-128, yes?